### ISSUE TITLE
Question about the encoder

### ISSUE THREAD

**Issue by wangzizhao** [Wuziyi616/SlotDiffusion#7]
Hi,

Thanks for sharing the code!

I have a quesiton about the encoder during slot extraction. IIUC, to extract slots, you use fist various encoder such as ResNet18, DINO ViT to get high-level features. Meanwhile, there is also a pretrained VQ-VAE used during decoding. I wonder what is the motivation to learn a separate encoder from scratch, instead of using the encoder of the pretrained VQ-VAE. Will using VQ-VAE's encoder cause any drawbacks?

Thanks a lot for your clarification in advance!

**Comment by Wuziyi616** [comment]
Hi, thanks for your interest in our work. The goal of VQ-VAE is to map the input image to a lower-dimensional latent space, which serves as the reconstruction target. Compared to directly reconstructing raw pixels, having a lower-dim latent space not only saves computation and memory but also leads to better learned slot disentanglement (as shown in our experiments and also the concurrent work LSD).

Re-using the VQ-VAE encoder is an interesting question, I haven't tried that. Maybe you can try to see how will it work.

My guess is that, the VQ-VAE is pre-trained in self-reconstruction, so it's biased towards low-level details like textures, i.e., it doesn't learn to extract high-level semantic information from input images. Instead, to extract object slots, we want the encoded visual features to contain more semantic information (e.g. object clustering). So VQ-VAE may underperform powerful encoders like DINO or ResNet.

Let me know if this answers your question.

**Comment by wangzizhao** [comment]
Thanks for the fast and detailed reply! That makes a lot sense.
